mautic/mautic
# Fix integration sync for deleted contacts

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Y]
| New feature/enhancement? (use the a.x branch)      | [Y]
| Deprecations?                          | [N]
| BC breaks? (use the c.x branch)        | [N]
| Automated tests included?              | [Y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Integration sync fails if record exist for deleted contact before syncing to Integration. This issue is not easily reproducible as it is mainly due to race condition.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Setup the integration which uses Integration Bundle.

Create a record in mautic_sync_object_field_change_report with following data.
integration = Salesforce2
object_id=any id which is not in leads
object_type=Mautic\LeadBundle\Entity\Lead
modified_at=
firstname=firstname
cloumn_type=string
cloumn_value=anystring

3. wait for integration_sync job to finish you will see no error and job gets completed.

4. You should see the added record is not deleted.

5. Execute command bin/console mautic:integrations:cleanup and you will see the record gets deleted. Ideally this command will executed daily.
